src/pages/index.js

```diff
@@ -11,86 +11,198 @@ import {
+            .then((res) => {
+              cardElement.handleDeleteButton(res);
+              confirmPreviewPopup.close();
+            })
+            .catch((err) => console.log(`An error occured: ${err}`))
+            .finally(() => confirmPreviewPopup.renderLoading(false));
+        });
       },
+      handleLike: () => {
+        if (!cardElement.isCardLiked()) {
+          return api
+            .likeCard(cardElement._cardId)
+            .then((res) => {
+              cardElement.setlikes(res.isLiked);
+            })
+            .catch((err) => {
+              console.error(err);
+            });
+        } else {
+          return api
+            .disLikeCard(cardElement._cardId)
+            .then((res) => {
+              cardElement.setlikes(res.isLiked);
+            })
+            .catch((err) => {
+              console.error(err);
+            });
+        }
+      },
+      userId: userInfo.getId(),
     },
     cardSettings.cardTemplate
   );
   return cardElement.getView();
 }
 
-const cardLayout = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
-      const cardEl = createCard(data);
-
-      cardLayout.addItem(cardEl);
-    },
-  },
-  cardSettings.cardList
-);
-cardLayout.renderItems();
+const renderCard = (data) => {
+  const cardEl = createCard(data);
+  cardSection.addItem(cardEl);
+};
 
 const userInfoPopup = new PopupWithForm({
   popupSelector: profileEditModal,
   handleFormSubmit: (userData) => {
-    userInfo.setUserInfo(userData);
-    userInfoPopup.close();
+    userInfoPopup.renderLoading(true);
+    api
+      .editProfileApi(userData)
+      .then((userData) => {
+        userInfo.setUserInfo(userData);
+        userInfoPopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => userInfoPopup.renderLoading(false));
   },
 });
 
 const newCardPopup = new PopupWithForm({
   popupSelector: addCardModal,
   handleFormSubmit: (data) => {
-    const card = createCard(data);
-    cardLayout.addItem(card);
-    newCardPopup.close();
+    newCardPopup.renderLoading(true);
+    api
+      .createCardApi(data)
+      .then((data) => {
+        renderCard(data);
+        newCardPopup.close();
+        addFormValidator.disableButton();
+      })
+      .catch((err) => console.log(`An error occured ${err}`))
+      .finally(() => newCardPopup.renderLoading(false));
+  },
+});
+
+const profileImagePopup = new PopupWithForm({
+  popupSelector: profileImageEdit,
+  handleFormSubmit: (data) => {
+    profileImagePopup.renderLoading(true);
+    api
+      .updateAvatar(data)
+      .then((data) => {
+        userInfo.setUserInfo(data);
+        profileImagePopup.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => profileImagePopup.renderLoading(false));
+  },
+});
+
+const confirmPreviewPopup = new PopupWithConfirmation({
+  popupSelector: cardDeletePopup,
+  handleFormSubmit: (id) => {
```

Comment:
Needs correction: Please remove this constructor argument, since it's not used, you have `setSubmitAction`